src/main/java/org/graylog/integrations/notifications/types/SlackEventNotification.java

```diff
@@ -35,148 +36,161 @@
+     * @param config
+     * @return
+     * @throws PermanentEventNotificationException - throws this exception when the custom message template is invalid
+     */
+    SlackMessage createSlackMessage(EventNotificationContext ctx, SlackEventNotificationConfig config) throws PermanentEventNotificationException {
+        //Note: Link names if notify channel or else the channel tag will be plain text.
+        boolean linkNames = config.linkNames() || config.notifyChannel();
+        String message = buildDefaultMessage(ctx, config);
+
+        String customMessage = null;
+        String template = config.customMessage();
+        if (!isNullOrEmpty(template)) {
+            customMessage = buildCustomMessage(ctx, config, template);
+        }
+
+        return new SlackMessage(
+                config.color(),
+                config.iconEmoji(),
+                config.iconUrl(),
+                config.userName(),
+                config.channel(),
+                linkNames,
+                message,
+                customMessage
+        );
+    }
+
+    String buildDefaultMessage(EventNotificationContext ctx, SlackEventNotificationConfig config) {
+        String title = buildMessageTitle(ctx);
+
+        // Build custom message
+        String audience = config.notifyChannel() ? "@channel " : "";
+        String description = ctx.eventDefinition().map(EventDefinitionDto::description).orElse("");
+        return String.format(Locale.ROOT, "%s*Alert %s* triggered:\n> %s \n", audience, title, description);
+    }
+
+    private String buildMessageTitle(EventNotificationContext ctx) {
+        String eventDefinitionName = ctx.eventDefinition().map(EventDefinitionDto::title).orElse("Unnamed");
+        return "_" + eventDefinitionName + "_";
+    }
+
+    String buildCustomMessage(EventNotificationContext ctx, SlackEventNotificationConfig config, String template) throws PermanentEventNotificationException {
+        final List<MessageSummary> backlog = getAlarmBacklog(ctx);
+        final List<MessageSummary> truncatedBacklog = getMessageBacklog(config, backlog);
+
+        Map<String, Object> model = getCustomMessageModel(ctx, config.type(), truncatedBacklog);
+        try {
+            LOG.debug("template = {} model = {}", template, model);
+            return templateEngine.transform(template, model);
+        } catch (Exception e) {
+            LOG.error("Exception during templating [{}]", e.toString());
+            throw new PermanentEventNotificationException(e.toString(), e.getCause());
+        }
+    }
+
+    @VisibleForTesting
+    List<MessageSummary> getMessageBacklog(SlackEventNotificationConfig config, List<MessageSummary> backlog) {
+        List<MessageSummary> truncatedBacklog = Collections.emptyList();
+        if (config.backlogSize() > 0 && backlog != null) {
+            truncatedBacklog = backlog.stream().limit(config.backlogSize()).collect(Collectors.toList());
+            return truncatedBacklog;
+        }
```

Comment:
No need to create a new variable here. You can just return the new one. Also, `Collections.emptyList()` returns an immutable list and thus shouldn't be used in contexts where it might be modified.

```suggestion
        if (config.backlogSize() > 0 && backlog != null) {
            return backlog.stream().limit(config.backlogSize()).collect(Collectors.toList());
        }
```